Create App Header

devimalka opened the pull request:
- add shadcn ui
- add npm scritps vite build watch
- refactor App.tsx for minimal config
- refactor alertHealthCheck function
- Add PRReviewHeader component
- refactor HealthCheckButton

**kept the health check button for now can refactor for later**

<img width="1910" height="242" alt="image" src="https://github.com/user-attachments/assets/f2a9b4ef-7d8c-498a-9ff6-6c412c9a6ff4" />
<img width="836" height="668" alt="image" src="https://github.com/user-attachments/assets/055f9672-583a-4567-94a8-9ee152338fc0" />

devimalka committed:
Create App Header

- add shadcn ui
- add npm scritps vite build watch
- refactor App.tsx for minimal config
- refactor alertHealthCheck function
- Add PRReviewHeader component
- refactor HealthCheckButton

LMgit91 commented on apps/client/src/App.tsx:
```diff
@@ -1,42 +1,45 @@
-      alert(
-        `health check response: ${response.status} - ${JSON.stringify(response.data)}`,
-      );
-    } catch {
```
You can't write a proper message handling for error

LMgit91 commented on apps/client/src/App.tsx:
```diff
@@ -1,42 +1,45 @@
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button type="button" onClick={() => alertHealthCheck}>
```
There is an issue alertHealthCheck should finish with"()", now nothing happen.

devimalka replied:
Hi think i think you are confusing the deleted lines with the added lines it's the previous code base the PR include a function HealthCheckButton with proper function calling

devimalka replied:
hi thank you pin pointing for that. i will update code base

LMgit91 reviewed (dismissed):
You can check I leave you some recommendations. If you can change it it will be good.

devimalka committed:
Fix alerHealthCheck function

devimalka committed:
Add error for alerHealthCheck console.error printing

LMgit91 reviewed (dismissed):
good job

actocodes commented on apps/client/tsconfig.app.json:
```diff
@@ -6,6 +6,10 @@
+      "@/*": ["./src/*"]
+    },
 
     /* Bundler mode */
```
remove comments in JSON

snowbytes commented on apps/client/eslint.config.js:
```diff
@@ -23,5 +23,6 @@ export default makeConfigFromBase(
         globals: globals.browser,
       },
     },
+    globalIgnores(["src/components/*"]),
```
We should make sure the vendored components from shadcn also pass our linting rules, as it sometimes impacts things like fast refresh (as is the case for exports of strings and components at the same time).

devimalka replied:
noted. will remove the globalIgnores and do the linting

snowbytes commented on apps/client/package.json:
```diff
@@ -9,20 +9,31 @@
     "lint": "eslint .",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "watch": "vite build --watch"
```
what do we use this for? AFAIK, we don't need to run a build script with `--watch` but run the dev script instead. Am I missing something?

snowbytes commented on apps/client/package.json:
```diff
@@ -9,20 +9,31 @@
     "react": "19.1.1",
-    "react-dom": "19.1.1"
+    "react-dom": "19.1.1",
+    "react-resizable-panels": "3.0.6",
```
what are we using this library for? It's not used anywhere

devimalka replied:
I'm not sure how it was added. will remove it

snowbytes commented on .githooks/pre-commit:
Thanks for fixing it! For some reason doing chmod on windows doesn't work sometimes

devimalka replied:
I think for the windows we need to specify the powershell path to execute the hook on windows. i'm not really sure about this

snowbytes replied:
It shouldn't be an issue as they're not meant to be executed manually. So long as [the prepare script](https://github.com/chingu-voyages/V57-tier3-team-32/blob/2d45598472c7b19c5b0b2c47636263387ae00e89/package.json#L15) got executed at least once (which would be the case if `npm ci` was run locally before, it'll set `.githooks/` as the hooks folder. Our hooks will then start running on committed files automatically.

snowbytes commented on apps/client/package.json:
I am not sure whether or not libraries like `class-variance-authority`, `clsx`, `tailwindcss` or `tailwind-merge` should be production libraries. It would be worthwhile to find out if they have any production presence. If not, we'll know to move them to `devDependencies`.

snowbytes replied:
TIL categorizing a dependency as dev vs prod has 0 effect on frontend code because bundlers like vite and webpack make their own dependency graphs. [Someone explained it better than me here](<https://jsramblings.com/do-dependencies-devdependencies-matter-when-using-webpack/>). So this doesn't really matter.

I'll leave it to you to resolve this conversation @devimalka.

snowbytes commented on apps/client/src/App.tsx:
```diff
@@ -1,42 +1,45 @@
+        variant="outline"
+        onClick={() => {
+          void (async () => {
+            await alertHealthCheck();
+          })();
+        }}
```
you don't need to create a async IIFE inside. a `onClick={() => { void alertHealthCheck() }}` is enough

snowbytes commented on apps/client/src/App.tsx:
```diff
@@ -1,42 +1,45 @@
-  };
   return (
-    <>
+    <div className="">
```
there's a dangling `className` here.

devimalka replied:
fixed

devimalka committed:
Merge branch 'dev' into chore/E1_US1_T1_app-header

snowbytes commented on apps/client/src/components/ui/button.tsx:
```diff
@@ -0,0 +1,58 @@
+  VariantProps<typeof buttonVariants> & {
+    asChild?: boolean;
+  }) {
+  const Comp = asChild ? Slot : "button";
```
This will raise linting errors after it's reenabled, but it's caused by a bug in our current version of @types/react. You can fix it by bumping the version to `19.1.15` (you can run `npm run i -D @types/react@19.1.15 -w apps/client` from the monorepo root)

snowbytes commented on apps/client/src/components/ui/button.tsx:
```diff
@@ -0,0 +1,58 @@
+  );
+}
+
+export { Button, buttonVariants };
```
This line will raise a linting error for fast refresh after you remove that `globalIgnores` in our eslint config, and for good reason. Component exports should be separate from string exports to minimize re-renders. The best approach is to transition to folder based components, then separate string exports from component exports. To do so, you'll need to:
- create a dedicated folder for the component (components/ui/button.tsx => components/ui/Button)
- make a `Button/index.tsx` that exports the component (`export function Button ...`)
- make a `Button/variants.ts` that exports the variants string (`export const variants ...`)

import format should not change if it was extensionless (e.g: `import { Button } from './components/ui/button'` => `import { Button } from './components/ui/Button'` will automatically import `Button/index.tsx`)

snowbytes commented on apps/client/src/components/ui/Card/index.tsx:
You're free to move this file to a dedicated Card folder or leave it as it is. But if you leave it as a file, please rename it to `Card.tsx` instead (capitalized) to follow naming conventions for frontend components

snowbytes commented on apps/client/src/Header.tsx:
Capitalization for naming conventions here too

snowbytes commented on apps/client/vite.config.ts:
```diff
@@ -1,7 +1,14 @@
+  plugins: [react(), tailwindcss()],
+  resolve: {
+    alias: {
+      "@": path.resolve(__dirname, "./src"),
```
You like your `@` aliases, I noticed you added them everywhere 😀

snowbytes reviewed (changes requested):
Hello, and thank you for your work! I noticed that you received reviews already, so I'll try not to mention what has been mentioned before.

That said, this time, it's necessary to rebase over the existing dev branch properly. If you haven't done so, please look at how to rebase and resolve conflicts while doing that. Here's [a decent video on the matter](https://www.youtube.com/watch?v=jhtbhSpV5YA) that @recklessbud shared in our team channel at the beginning of the voyage.

I'm looking forward to seeing the changes and/or hearing back from you about what was brought up here. 😀

devimalka committed:
Fix Css issues with merge conflict

 - remove dark background
 - remove flex
 - remove center items

devimalka committed:
Minor Fixes

- remove npm run watch command
- remove the esling globalIgnores

devimalka committed:
Fix unused Dependencies

devimalka committed:
Fix linting Issue

 - Separate the button logic components/ui/button.tsx to components/ui/Button folder
 - components/ui/Button/index.tsx => Button Component
 - components/ui/Button/variants.tsx => Button Variants

devimalka committed:
set pre-push hook file to chmod+x

devimalka committed:
Capitalization for naming conventions

snowbytes commented on apps/client/package.json:
```diff
@@ -12,19 +12,26 @@
   "devDependencies": {
     "@tailwindcss/vite": "4.1.13",
-    "@types/react": "19.1.10",
+    "@types/node": "24.5.2",
```
I don't see `@types/node` used anywhere. Is it for `__dirname` in `vite.config.ts`? if that's the case, you can use `fileURLToPath(import.meta.url)` or the newer `import.meta.dirname` instead

devimalka replied:
actually i'm not sure about this i'll check

snowbytes commented on apps/client/src/components/ui/Button/index.tsx:
```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
```
The import is only used for types, it should be changed to `import type { ComponentProps } from 'react'`, or like you're doing for `VariantProps` below. Generally we avoid namespace imports (`import * as ...`) because it can't be tree shaked

snowbytes commented on apps/client/src/components/ui/Button/variants.ts:
the file's extension should be changed to `.ts` since it's not a react component

snowbytes commented on apps/client/src/Header.tsx:
```diff
@@ -0,0 +1,43 @@
+import { Card, CardContent } from "./components/ui/card";
+
+function CurrentDate() {
+  const currentDate = new Date();
+  const days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+  const months = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+  ];
+  const day = days[currentDate.getDay()];
+  const month = months[currentDate.getMonth()];
+  const date = currentDate.getDate();
+  const year = currentDate.getFullYear();
+
+  return `${day}, ${date} ${month} ${year}`;
+}
```
This is already part of the standard lib as [`Date.prototype.toDateString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toDateString). By the way, the convention in js is to only use PascalCase for object constructors and react component functions. For all other cases, we use camelCase.

devimalka replied:
agree with that but it won't return the  except output that's why i have to use it.

snowbytes replied:
you mean the extra comma? You can just do `(new Date()).toDateString().replace(" ", ", ")`

snowbytes commented on apps/client/src/Header.tsx:
```diff
@@ -0,0 +1,43 @@
+            PR Review Status Board
+          </h2>
+          <span className="text-xs sm:text-sm lg:text-base text-slate-500">
+            <CurrentDate></CurrentDate>
```
FYI, since this is a function that returns a string and not a react component, you should [call it directly](https://react.dev/learn/javascript-in-jsx-with-curly-braces#using-curly-braces-a-window-into-the-javascript-world)

snowbytes commented on apps/client/vite.config.ts:
```diff
@@ -1,8 +1,15 @@
   plugins: [react(), tailwindcss()],
+  resolve: {
+    alias: {
+      "@": path.resolve(__dirname, "./src"),
```
we shouldn't rely on `__dirname` being present since we're using esm. prefer using `import.meta.dirname` or the older `fileURLToPath(import.meta.url)`

snowbytes commented on package-lock.json:
Seems like something wrong happened with installs, seeing as there are almost 4k deletions... I'll take care of it separately afterwards.

snowbytes commented on apps/client/src/App.tsx:
```diff
@@ -1,42 +1,44 @@
+  return (
+    <div className="block">
+      <Button
+        variant="outline"
```
Like I said in our coms channel, the variants aren't properly applied and should be fixed

devimalka replied:
found the issue. previous commit i have messed with the index.css file now it's fine i will update code base

snowbytes reviewed (changes requested):
As requested, here's a second review. I noticed that 3 comments in my previous review were not addressed, please check them out.
Good luck!

devimalka committed:
change variants.tsx to variants.ts

devimalka committed:
Remove @types/node from the package.json

devimalka committed:
Fix Button variants not working

 - fixed button variants not working
 - fixed the calling of currentDate function to function call

devimalka committed:
Refact the function

 - Refactor the function CurrentDate to camelCase => currentDate

snowbytes commented on apps/client/src/index.css:
```diff
@@ -1,70 +1,120 @@
+  --color-accent: var(--accent);
+  --color-accent-foreground: var(--accent-foreground);
+  --color-destructive: var(--destructive);
+  --color-border: var(--border);
```
Looking at the getting started docs for shadcn I think you're missing a `--color-destructive-forground: var(--destructive-foreground);`. Then again, the oklch values seem to be different. Was this generated by a command or copy pasted from somwhere? Just making sure. I might be totally wrong here 😅

devimalka replied:
it was generated with tailwind command. i have messed with the previous index.css file and removed the css code. had to regenerate it

snowbytes commented on apps/client/tsconfig.json:
```diff
@@ -3,5 +3,11 @@
+    "paths": {
+      "@/*": ["./src/*"]
+    }
+  }
```
you don't need this here. Since you already have it in `tsconfig.app.json` where it's needed and referencing it here 👍

devimalka replied:
so you want me to remove the lines?

devimalka replied:
removed the lines and pushed the commit to the the PR

devimalka committed:
Remove apps/client/tsconfig.json lines

devimalka committed:
Fix PR Reviews

 - fixed the PR Review
 - Change CurrrentDate function
 - remove the lines in tsconfig.json file
 - remove comments

devimalka committed:
Regenerate package-lock.json

snowbytes reviewed (approved):
Great work pushing through! LGTM
